app/test/e2e/layer-update.spec.js

```diff
@@ -0,0 +1,132 @@
+        const datasetLayer = await updateLayer({ role: ROLES.USER });
+        datasetLayer.status.should.equal(403);
+        ensureCorrectError(datasetLayer.body, 'Forbidden');
+    });
+
+    it('PATCH /dataset/:dataset/layer/:layer - should return error not found, when layer is not exist', async () => {
+        createMockDataset('123');
+        const layer = createLayer(['rw'], '123', '321');
+        await new Layer(layer).save();
+
+        const datasetLayer = await requester
+            .delete(`${datasetPrefix}/123/layer/123?loggedUser=${JSON.stringify(ROLES.ADMIN)}`)
+            .send();
+
+        datasetLayer.status.should.equal(404);
+        ensureCorrectError(datasetLayer.body, 'Layer with id \'123\' doesn\'t exist');
+    });
+
+    it('PATCH /dataset/:dataset/layer/:layer - Update specific layer with wrong data, should fail', async () => {
+        await Promise.all(WRONG_DATAS.map(async ({ data, expectedError }) => {
+            const datasetLayer = await updateLayer({ role: ROLES.ADMIN }, data);
+            datasetLayer.status.should.equal(400);
+            ensureCorrectError(datasetLayer.body, expectedError);
+        }));
+    });
+
+    it('PATCH /dataset/:dataset/layer/:layer - Update specific layer should update', async () => {
+        const datasetLayer = await updateLayer({ role: ROLES.ADMIN }, LAYER_TO_UPDATE);
+        datasetLayer.status.should.equal(200);
+        const responseData = datasetLayer.body.data;
+
+        Object.keys(LAYER_TO_UPDATE)
+            .map(field => responseData.attributes[field].should.deep.equal(LAYER_TO_UPDATE[field]));
+    });
+
+    it('PATCH /layer/change-environment/:dataset/:env - should return error not found, when dataset is not exist', async () => {
+        const envLayer = await requester.patch(`${layerPrefix}/change-environment/123/test`);
+        envLayer.status.should.equal(404);
+        ensureCorrectError(envLayer.body, 'Dataset not found');
+    });
+
+    it('PATCH /layer/change-environment/:dataset/:env - Update layer env without being authenticated should fail', async () => {
+        const envLayer = await updateEnv({ role: null });
+        envLayer.status.should.equal(403);
+        ensureCorrectError(envLayer.body, 'Not authorized');
+    });
+
+    it('PATCH /layer/change-environment/:dataset/:env - Update layer env with being authenticated not as MICROSERVICE should fail', async () => {
+        const envLayer = await updateEnv({ role: ROLES.ADMIN });
+        envLayer.status.should.equal(403);
+        ensureCorrectError(envLayer.body, 'Not authorized');
+    });
+
+    it('PATCH /layer/change-environment/:dataset/:env - Should update layer env', async () => {
+        const layerId = getUUID();
+
+        const envLayer = await updateEnv({ role: ROLES.MICROSERVICE, layerId });
+        envLayer.status.should.equal(200);
+        const layer = await Layer.findOne({ _id: layerId });
+        expect(layer.env).to.be.equal('test');
```

Comment:
is there a response body? if so, please test it

Reply:
there is an empty body